src/System.ComponentModel.Composition/tests/Microsoft/Internal/AssumesTests.cs

```diff
@@ -0,0 +1,252 @@
+        }
+
+        [Fact]
+        public void NotNullOrEmpty_EmptyAsValueArgument_ShouldThrowInternalErrorException()
+        {
+            Throws(() =>
+            {
+                Assumes.NotNullOrEmpty("");
+            });
+        }
+
+        [Fact]
+        public void NotNullOrEmpty_ValueAsValueArgument_ShouldNotThrow()
+        {
+            var expectations = new List<string>();
+            expectations.Add(" ");
+            expectations.Add("  ");
+            expectations.Add("   ");
+            expectations.Add("Value");
+
+            foreach (var e in expectations)
+            {
+                Assumes.NotNullOrEmpty(e);
+            }
+        }
+
+        [Fact]
+        public void IsTrue1_FalseAsConditionArgument_ShouldThrowInternalErrorException()
+        {
+            Throws(() =>
+            {
+                Assumes.IsTrue(false);
+            });
+        }
+
+        [Fact]
+        public void IsTrue2_FalseAsConditionArgument_ShouldThrowInternalErrorException()
+        {
+            Throws(() =>
+            {
+                Assumes.IsTrue(false, "Message");
+            });
+        }
+
+        [Fact]
+        public void IsTrue1_TrueAsConditionArgument_ShouldNotThrow()
+        {
+            Assumes.IsTrue(true);
+        }
+
+        [Fact]
+        public void IsTrue2_TrueAsConditionArgument_ShouldNotThrow()
+        {
+            Assumes.IsTrue(true, "Message");
+        }
+
+        [Fact]
+        public void NotReachable_ShouldAlwaysThrow()
+        {
+            Throws(() =>
+            {
+                Assumes.NotReachable<object>();
+            });
+        }
+
+        private static void Throws(Action action)
+        {
+            try
+            {
+                action();
+                Assert.False(true);
+            }
+            catch (Exception ex)
+            {
+                Type exceptionType = ex.GetType();
+
+                // The exception should not be a 
+                // publicily catchable exception
+                Assert.False(exceptionType.IsVisible);
+            }
+        }
+
+#if FEATURE_SERIALIZATION
+        [Fact]
+        public void Message_CanBeSerialized()
+        {
+            var expectations = Expectations.GetExceptionMessages();
+
+            foreach (var e in expectations)
+            {
+                var exception = CreateInternalErrorException(e);
+
+                var result = SerializationTestServices.RoundTrip(exception);
+
+                Assert.AreEqual(exception.Message, result.Message);
+            }
+        }
+
+#endif //FEATURE_SERIALIZATION
+
+        private static Exception CreateInternalErrorException()
+        {
+            return CreateInternalErrorException((string)null);
+        }
+
+        private static Exception CreateInternalErrorException(string message)
+        {
+            Exception exception = null;
+
+            try
+            {
+                Assumes.IsTrue(false, message);
```

Comment:
This whole method could be rewritten to:

```cs
Exception e = Assert.ThrowsAny<Exception>(() => Assumes.IsTrue(false, message));
return e;
```

If the exception is null, or not thrown the Assert.ThrowsAny will fail, that way you don't have to validate if it is not null.

Reply:
or even better it could be a one liner method: 

```cs
private static Exception CreateInternalErrorException(string message) => Assert.ThrowsAny<Exception>(() => Assumes.IsTrue(false, message));
```